rust/parquet/src/arrow/schema.rs

```diff
@@ -42,14 +42,10 @@ pub fn parquet_to_arrow_schema(
     key_value_metadata: &Option<Vec<KeyValue>>,
 ) -> Result<Schema> {
     let mut metadata = parse_key_value_metadata(key_value_metadata).unwrap_or_default();
-    let maybe_schema = metadata
+    let arrow_schema_metadata = metadata
         .remove(super::ARROW_SCHEMA_META_KEY)
-        .map(|encoded| get_arrow_schema_from_metadata(&encoded));
-
-    let arrow_schema_metadata = match maybe_schema {
-        Some(v) => Some(v?),
-        _ => None,
-    };
+        .map(|encoded| get_arrow_schema_from_metadata(&encoded))
+        .map_or(Ok(None), |v| v.map(Some))?;
```

Comment:
I think this can use unwrap_or, as both of the cases map to Ok?

Reply:
@Dandandan thanks a lot!  The code looks more concise!